Education-analysis.sql

```diff
@@ -0,0 +1,292 @@
+/*1)	Gross intake ratio in first grade of primary education, female (% of relevant age group)*/
+create temp table Gross_intake_1grade_female
+as
+select 	distinct i.countryname
+,		i."Year" 
+,		i.value
+,		round(lag(i.value) over (partition by i.countryname order by i."Year")::numeric,2) previous_year
+,		round(((i.value - lag(i.value) over (partition by i.countryname order by i."Year"))/
+		lag(i.value) over (partition by i.countryname order by i."Year"))::numeric,2) YoY_value
+from indicators i
+join country c on i.countrycode = c.countrycode 
+where c.region is not null and c.region <> '' and indicatorcode = 'SE.PRM.GINT.FE.ZS'
+group by i.countryname, i."Year", i.value;
+
+select * from Gross_intake_1grade_female;
+
+/*2)	Gross intake ratio in first grade of primary education, male (% of relevant age group)*/
+create temp table Gross_intake_1grade_male
+as
+select 	distinct i.countryname
+,		i."Year" 
+,		i.value
+,		round(lag(i.value) over (partition by i.countryname order by i."Year")::numeric,2) previous_year
+,		round(((i.value - lag(i.value) over (partition by i.countryname order by i."Year"))/
```

Comment:
kiedy wyliczamy dużo takich zmiennych pomocniczych, to warto stworzyć dodatkową tabelę pośrednią, w której jedynie wyliczamyt e wielkości, a potem dopiero w kolejnych krokach na bazie tej tabeli dzielimy zmienne przez siebie, aby uniknąć pisania tak długich dzieleń, które mocno zaciemniają kod.